File: BackendAPI/src/main/java/com/aguare/backendapi/Entitys/Sale.java
package com.aguare.backendapi.Entitys;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;
import java.sql.Date;

@Data
@Entity
@Table(name = "sale")
public class Sale implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idSale;
    private Date date;
    private Double total;

    @Column(name = "fk_id_branch")
    private Long idBranch;

    @Column(name = "fk_dpi")
    private Long dpi;

    @Column(name = "fk_nit")
    private String nit;
}

File: FrontApp/src/app/Entitys/Sale.ts
export class Sale {
  idSale: number;
  date: Date;
  total: number;
  idBranch: number;
  dpi: string;
  nit: string;

  constructor(
    idSale: number,
    date: Date,
    total: number,
    idBranch: number,
    DPI: string,
    NIT: string
  ) {
    this.idSale = idSale;
    this.date = date;
    this.total = total;
    this.idBranch = idBranch;
    this.dpi = DPI;
    this.nit = NIT;
  }
}
